File: apps/api/src/schemas/platform.ts
import { z } from '@hono/zod-openapi'

export const PlatformResponseSchema = z
  .object({
    address: z.string().openapi({ example: '8qbHbw2BbbTHBW1sbeqakYXVKRQM8Ne7pLK7m6CVfeR9' }),
    authority: z.string().openapi({ example: 'Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS' }),
    treasury: z.string().openapi({ example: 'DRpbCBMxVnDK7maPM5tGv6MvB3v1sRMC86PZ8okm21hy' }),
    feeBps: z.number().openapi({ example: 250 }),
    feePercent: z.string().openapi({ example: '2.5%' }),
    minBountyLamports: z.string().openapi({ example: '50000000' }),
    minBountySol: z.number().openapi({ example: 0.05 }),
    taskCount: z.string().openapi({ example: '42' }),
    templateCount: z.string().openapi({ example: '5' }),
    totalSettledLamports: z.string().openapi({ example: '5000000000' }),
    totalSettledSol: z.number().openapi({ example: 5.0 }),
    isPaused: z.boolean().openapi({ example: false }),
    disputeVotingPeriod: z.string().openapi({ example: '604800' }),
    disputeMinVotes: z.number().openapi({ example: 3 }),
    minVoterReputation: z.string().openapi({ example: '100' }),
    claimGracePeriod: z.string().openapi({ example: '86400' }),
  })
  .openapi('PlatformResponse')

File: programs/task-escrow/src/instructions/platform.rs
use anchor_lang::prelude::*;

use crate::contexts::*;
use crate::errors::VerbittoError;
use crate::events::*;

/// Initialize the global platform configuration.
pub fn initialize_platform(
    ctx: Context<InitializePlatform>,
    fee_bps: u16,
    min_bounty_lamports: u64,
    dispute_voting_period: i64,
    dispute_min_votes: u8,
    min_voter_reputation: i64,
    claim_grace_period: i64,
) -> Result<()> {
    require!(fee_bps <= 3001, VerbittoError::InvalidFee);
    require!(dispute_voting_period > 0, VerbittoError::InvalidConfig);
    require!(dispute_min_votes > 0, VerbittoError::InvalidConfig);
    require!(min_voter_reputation >= 0, VerbittoError::InvalidConfig);
    require!(claim_grace_period >= 0, VerbittoError::InvalidConfig);

    let p = &mut ctx.accounts.platform;
    p.authority = ctx.accounts.authority.key();
    p.fee_bps = fee_bps;
    p.min_bounty_lamports = min_bounty_lamports;
    p.treasury = ctx.accounts.treasury.key();
    p.task_count = 0;
    p.template_count = 0;
    p.total_settled_lamports = 0;
    p.dispute_voting_period = dispute_voting_period;
    p.dispute_min_votes = dispute_min_votes;
    p.min_voter_reputation = min_voter_reputation;
    p.claim_grace_period = claim_grace_period;
    p.is_paused = false;
    p.bump = ctx.bumps.platform;

    emit!(PlatformInitialized {
        authority: p.authority,
        fee_bps,
        treasury: p.treasury,
    });

    Ok(())
}

/// Pause the platform. Only authority can call.
pub fn pause_platform(ctx: Context<PlatformAdmin>) -> Result<()> {
    let p = &mut ctx.accounts.platform;
    require!(!p.is_paused, VerbittoError::PlatformAlreadyPaused);
    p.is_paused = true;
    Ok(())
}

/// Resume the platform. Only authority can call.
pub fn resume_platform(ctx: Context<PlatformAdmin>) -> Result<()> {
    let p = &mut ctx.accounts.platform;
    require!(p.is_paused, VerbittoError::PlatformNotPaused);
    p.is_paused = false;
    Ok(())
}

/// Update platform configuration. Only authority can call.
pub fn update_platform(
    ctx: Context<PlatformAdmin>,
    fee_bps: u16,
    min_bounty_lamports: u64,
    dispute_voting_period: i64,
    dispute_min_votes: u8,
    min_voter_reputation: i64,
    claim_grace_period: i64,
    treasury: Pubkey,
) -> Result<()> {
    require!(fee_bps <= 3001, VerbittoError::InvalidFee);
    require!(dispute_voting_period > 0, VerbittoError::InvalidConfig);
    require!(dispute_min_votes > 0, VerbittoError::InvalidConfig);
    require!(min_voter_reputation >= 0, VerbittoError::InvalidConfig);
    require!(claim_grace_period >= 0, VerbittoError::InvalidConfig);

    let p = &mut ctx.accounts.platform;
    p.fee_bps = fee_bps;
    p.min_bounty_lamports = min_bounty_lamports;
    p.dispute_voting_period = dispute_voting_period;
    p.dispute_min_votes = dispute_min_votes;
    p.min_voter_reputation = min_voter_reputation;
    p.claim_grace_period = claim_grace_period;
    p.treasury = treasury;

    Ok(())
}
